validator/src/main/java/com/google/daq/mqtt/registrar/Registrar.java

```diff
@@ -111,37 +112,11 @@ public class Registrar {
   public static void main(String[] args) {
     ArrayList<String> argList = new ArrayList<>(List.of(args));
     Registrar registrar = new Registrar();
-    executeWithRegistrar(registrar, argList);
+    processArgs(argList, registrar);
+    registrar.execute();
   }
 
-  public static void executeWithRegistrar(Registrar registrar, ArrayList<String> argList) {
-    try {
-      boolean processAllDevices = processArgs(argList, registrar);
-
-      if (registrar.schemaBase == null) {
-        registrar.setToolRoot(null);
-      }
-
-      registrar.loadSiteMetadata();
-
-      if (processAllDevices) {
-        registrar.processDevices();
-      } else {
-        registrar.processDevices(argList);
-      }
-
-      registrar.writeErrors();
-      registrar.shutdown();
-    } catch (ExceptionMap em) {
-      ExceptionMap.format(em, ERROR_FORMAT_INDENT).write(System.err);
-      throw new RuntimeException("mapped exceptions", em);
-    } catch (Exception ex) {
-      ex.printStackTrace();
-      throw new RuntimeException("main exception", ex);
-    }
-  }
-
-  private static boolean processArgs(List<String> argList, Registrar registrar) {
+  public static void processArgs(List<String> argList, Registrar registrar) {
```

Comment:
Still not sure why this is static?  The general adage with static methods for Java is to avoid them unless they need to be static for some reason. I think this comes down to the issue that you can't mock/override a static method, so it makes future testing/extension a problem sometimes. Not likely in this case, but just on general practice.  E.g., I would expect this to be just a regular method on the Registrar class (no need to pass in the explicit parameter).  Anyway, that's mainly the Java-style that I've always used, but it's been so long that I'm not 100% sure where it came from. I checked the Java style guide and it doesn't say anything about it.  Doesn't really matter -- I also optimize for "least number of words" and if you made it a class method, all references to "registrar." go away reducing the overall footprint of the class -- and there's no world in which you would call this method without a registrar class!